### ISSUE TITLE
[bug] Incorrect DPI scaling on Wayland/Linux

### ISSUE THREAD

**Issue by happenslol** [tauri-apps/tauri#5600]
### Describe the bug

Running a freshly initialized tauri project on a 2256x1504 Laptop produces the following window when running `yarn tauri dev`:

![2022-11-09_21-34_1](https://user-images.githubusercontent.com/753308/200936571-b6e47216-937f-484d-84f2-96447a8834fc.png)

Opening the URL of the local dev server in chrome with the ozone platform enabled shows the following:

![2022-11-09_21-34](https://user-images.githubusercontent.com/753308/200936706-8458e1fb-861c-4dcf-9409-ceaab8b2ff87.png)

It seems like the webview is not doing any scaling at all. Setting `GDK_SCALE` or `GDK_DPI_SCALE` does not have any effect either.

### Reproduction

1. Run `cargo-create-tauri-app`
2. Run `yarn tauri dev` on a HiDPI-screen

### Expected behavior

The scaling should be the same in the GTK WebView as in a correctly scaled Chromium or Firefox window.

### Platform and versions

Running on NixOS with Sway/Wayland and scaling set to 1.5, using the recommended `flake.nix`.

```
$ tauri info

Environment
  › OS: NixOS 22.11.0 X64
  › Node.js: 18.9.1
  › npm: 8.19.1
  › pnpm: Not installed!
  › yarn: 1.22.19
  › rustup: 1.25.1
  › rustc: 1.64.0
  › cargo: 1.64.0
  › Rust toolchain: stable-x86_64-unknown-linux-gnu

Packages
  › @tauri-apps/cli [NPM]: 1.2.0
  › @tauri-apps/api [NPM]: 1.2.0
  › tauri [RUST]: 1.2.0,
  › tauri-build [RUST]: 1.2.0,
  › tao [RUST]: 0.15.3,
  › wry [RUST]: 0.22.0,

App
  › build-type: bundle
  › CSP: unset
  › distDir: ../dist
  › devPath: http://localhost:1420/
  › framework: React

App directory structure
  ├─ public
  ├─ node_modules
  ├─ src-tauri
  ├─ .vscode
  ├─ .direnv
  └─ src
```

**Comment by happenslol** [comment]
Does anyone have any input on this? Is this maybe a better fit for another tauri-apps repository, since it deals more directly with the windowing/webkit2gtk layer? I'd also be open to digging into this if anyone could point me in the right direction.

I'd really love to use tauri for a project of mine, but this is really standing in the way of that.

**Comment by wusyong** [comment]
Is website with any URL link also affected? I've auto scaled the gtk window iirc.
Search a bit on webkit2gtk and I [found](https://bugs.eclipse.org/bugs/show_bug.cgi?id=525957) it might need to do it manually.
I can test this in wry if there's a minimum example to reproduce.

**Comment by elibroftw** [comment]
![image](https://github.com/tauri-apps/tauri/assets/21298211/afbab83c-787b-4826-ada0-eaea9a729a67)

User has 200% scaling, cannot scroll

**Comment by AaronHessler** [comment]
Experiencing the same issue here.
Any update on the situation?

**Comment by AaronHessler** [comment]
I think this might not be an issue related to tauri, but nixos instead.
I also happen to use nixos (btw).

**Comment by ryadios** [comment]
I am also experiencing the same issue. Using endeavour os with hyprland

**Comment by Toldoven** [comment]
**Found a workaround on NixOS + Niri + WebKitGTK 2.52.4**

Add this to your development environment variables:

```
GSETTINGS_SCHEMA_DIR = pkgs.glib.getSchemaPath pkgs.gsettings-desktop-schemas;
```

I'm using [devenv](https://devenv.sh/) to only add this to my development environment, but you can also add this to `home.sessionVariables`

**TLDR** 

Not a Tauri issue, caused by upstream dependencies (WebKitGTK, GTK)

Can confirm that it's likely an issue specific to NixOS/Guix-style environments

Also seems like it's a development environment only issue (shouldn't be an issue with packaged apps) 

**Root cause**

- The first hint was `window.devicePixelRatio`. The value is `-0.02083333395421505` when on my display it should be `2`
- There's a WebKitGTK bug: [`xftDPI()`](https://github.com/WebKit/WebKit/blob/a36097add8b01a89a6cac53320fb09eec72f24ff/Source/WebKit/UIProcess/gtk/SystemSettingsManagerProxyGtk.cpp#L113-L127) doesn't handle the documented "unset" sentinel (-1) from `gdk_screen_get_resolution()`. The value ends up in the page zoom factor, which is one of the multipliers in `window.devicePixelRatio`
- There is another bug in GTK: [`init_settings()`](https://github.com/GNOME/gtk/blob/cf0dd9763b4eca643bcfdf6ee40f2f39dfde9c33/gdk/wayland/gdkscreen-wayland.c#L824-L828) returns early when no GSettings schemas exist at all, skipping the code that would have initialized the DPI to its default (96). To clarify, it's not a specific schema we're missing - this only happens when there isn't a single valid schema
- Other distros don't have this issue because there's practically always at least one schema installed globally (GTK itself ships some). These are two upstream bugs that NixOS just happens to be the only place to surface
- The workaround is to make at least one GSettings schema discoverable in the development environment, so we don't fail this check

I'll file a more detailed report on this soon, and hopefully the issue can be resolved upstream
